src/session.rs

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::{Contextual, Result};
+
+use ::sev::certs::sev::Certificate;
+use ::sev::certs::Chain;
+use ::sev::launch;
+use ::sev::launch::sev::Policy;
+use ::sev::session;
+
+use std::convert::{From, TryFrom};
+use std::fs;
+use std::mem::size_of;
+use std::path::PathBuf;
+use std::slice::from_raw_parts;
+
+use codicon::{Decoder, Encoder};
+
+pub fn cmd(name: Option<String>, chain: PathBuf, policy: u32) -> Result<()> {
+    let (tik_f, tek_f, godh_f, session_f) = file_names(name);
+
+    let session = session::Session::try_from(Policy::from(policy))
+        .context("could not derive session from policy")?;
+
+    let mut tik_file = fs::File::create(tik_f).context("TIK file could not be created")?;
+    let mut tek_file = fs::File::create(tek_f).context("TEK file could not be created")?;
+
+    session
+        .tik
+        .encode(&mut tik_file, ())
+        .context("TIK could not be encoded into file")?;
+    session
+        .tek
+        .encode(&mut tek_file, ())
+        .context("TEK could not be encoded into file")?;
+
+    let mut chain_file = fs::File::open(chain).context("could not open certificate file")?;
+    let chain = Chain::decode(&mut chain_file, ())
+        .context("could not decode the given certificate chain")?;
+
+    let start = session
+        .start(chain)
+        .context("could not start session based off of provided certificate chain")?;
+
+    let session = start.session;
+    let session = unsafe {
+        from_raw_parts(
+            &session as *const launch::sev::Session as *const u8,
+            size_of::<launch::sev::Session>(),
+        )
+    };
+    let session = base64::encode(session);
+
+    let godh = start.cert;
+    let godh = unsafe {
+        from_raw_parts(
+            &godh as *const Certificate as *const u8,
+            size_of::<Certificate>(),
+        )
+    };
+    let godh = base64::encode(godh);
+
+    fs::write(session_f, session).context("could not write base64 encoded session")?;
+    fs::write(godh_f, godh).context("could not write base64 encoded godh")?;
+
+    Ok(())
+}
+
+fn file_names(name: Option<String>) -> (String, String, String, String) {
+    let prefix = match name {
+        Some(n) => n,
+        None => "vm".to_string(),
+    };
+
+    let tik = prefix.clone() + "_tik.bin";
+    let tek = prefix.clone() + "_tek.bin";
+    let godh = prefix.clone() + "_godh.bin";
+    let session = prefix + "_session.bin";
```

Comment:
The 2nd two files here are containing  base64 data rather than raw bytes like the first two, so I think `.bin` is a misleading extension - `.base64` or .`b64` is more appropriate

Reply:
Yea, `.b64` is a better option.